### PR TITLE
Added HTTP Light documentation.

### PR DESCRIPTION
**Pull request by Priva28**
**Description:**
Adds documentation of the HTTP Light component. See more information on my other pull request.
MQTT JSON Document was not changed. It was accidentally opened and had one character deleted which was then replaced.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#15354

## Checklist:

- [x] Branch: Fixes, changes and adjustments should be created against `current`. New documentation for platforms/components and features should go to `next`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/light.httplight.markdown [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: httplight.png
+ha_category: Light
+ha_release: 0.73
+ha_iot_class: "Local Polling"
+---
+
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: Remove the extra line.

File: source/_components/light.httplight.markdown [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+title: "HTTP Light"
+description: "Instructions on how to setup the HTTP light component within Home Assistant."
+date: 2018-7-07 10:00
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: httplight.png
+ha_category: Light
+ha_release: 0.73
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: `0.73` has already been shipped. Please update this to `0.74`

File: source/_components/light.httplight.markdown [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+```yaml
+# Example configuration.yaml entry
+light:
+  - platform: httplight
+    name: Light
+    host: 10.1.1.1
+    type: color
+```
+
+Configuration variables:
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: Please use the configuration tags, for more information please see:
https://home-assistant.io/developers/documentation/create_page/#configuration

File: source/_components/light.httplight.markdown [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+- **name** (*Required*): Name of the light.
+- **host** (*Required*): The IP Address of the light.
+- **type** (*Required*): The type of HTTP Light. Valid options are `color`, `dimmable` and `simple`.
+
+## {% linkable_title Different Light Types %}
+
+### {% linkable_title Color %}
+
+If you have a HTTP Light that supports color use the color type.
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: `a HTTP` -> `an HTTP`

File: source/_components/light.httplight.markdown [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+The HTTP Light component uses hex values to communicate color. Your device should be configured to return a hex value when `http://IP_ADDRESS/color` is called. Your device should change it's color when it recieves `http://IP_ADDRESS/set/HEX_VALUE_HERE`.
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: `recieves` -> `receives`

File: source/_components/light.httplight.markdown [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+# Example configuration.yaml entry for HTTP Light with color.
+light:
+- platform: httplight
+  name: Color Light
+  host: IP_ADDRESS
+  type: color
+```
+### {% linkable_title Dimmable %}
+
+If you have a HTTP Light that supports brightness but not color use the dimmable type.
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: `a HTTP` -> `an HTTP`

File: source/_components/light.httplight.markdown [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+Your device should be configured to return a brightness value from 0-100 when `http://IP_ADDRESS/brightness` is called. Your device should change the lights brightness when it recieves `http://IP_ADDRESS/set/BRIGHTNESS_HERE`.
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: - `the lights brightness` -> `the brightness of the light`
- `recieves` -> `receives`

File: source/_components/light.httplight.markdown [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+# Example configuration.yaml entry for HTTP Light that is dimmable.
+light:
+- platform: httplight
+  name: Dimmable Light
+  host: IP_ADDRESS
+  type: color
+```
+### {% linkable_title Simple %}
+
+If you have a HTTP Light that only supports `ON/OFF` commands, use the simple type.
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: `a HTTP` -> `an HTTP`

File: source/_components/light.httplight.markdown [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+Your device should be configured to return either `on` or `off` when `http://IP_ADDRESS/state` is called. Your device should turn on when it recieves `http://IP_ADDRESS/on`, and off when it recieves `http://IP_ADDRESS/off`
```
Comment (frenck) [inline source/_components/light.httplight.markdown]: `recieves` -> `receives` (2x)

File: source/_components/light.mqtt_json.markdown [hunk 10]
```diff
@@ -262,4 +262,4 @@ Home Assistant expects the hue values to be in the range 0 to 360 and the satura
...
 
-- [AiLight](https://github.com/stelgenhof/AiLight) is a custom firmware for the Ai-Thinker (and equivalent) RGBW WiFi light bulbs that has an ESP8266 onboard and controlled by the MY9291 LED driver. It implements the [MQTT JSON light](/components/light.mqtt_json/) platform and supports ON/OFF, RGBW colours, brightness, colour temperature, flashing and transitions. Also it includes [MQTT Auto Discovery](/docs/mqtt/discovery/)) and the MQTT Last Will and Testament is enabled as well.
\ No newline at end of file
+- [AiLight](https://github.com/stelgenhof/AiLight) is a custom firmware for the Ai-Thinker (and equivalent) RGBW WiFi light bulbs that has an ESP8266 onboard and controlled by the MY9291 LED driver. It implements the [MQTT JSON light](/components/light.mqtt_json/) platform and supports ON/OFF, RGBW colours, brightness, colour temperature, flashing and transitions. Also it includes [MQTT Auto Discovery](/docs/mqtt/discovery/)) and the MQTT Last Will and Testament is enabled as well.
```
Comment (frenck) [inline source/_components/light.mqtt_json.markdown:265]: I'm not sure why this is a change and how it is related to this PR?

**Review by frenck — CHANGES_REQUESTED** [review]
Thank you for the PR @Priva28!
I've reviewed it and some a couple of small things.
Could you take a look? 👍

**Comment by frenck** [comment]
Additionally, the added image is really big! Maybe optimize it? (e.g., using ImageOptim).

File: source/_components/light.mqtt_json.markdown (hunk 10 above)
Comment (Priva28) [reply source/_components/light.mqtt_json.markdown:265]: See my description. It was an accidentally opened when I was testing but restored to what it was before.

Comment (frenck) [reply source/_components/light.mqtt_json.markdown:265]: I've missed that, thanks for the clarification!

**Review by frenck — APPROVED** [review]
Thanks for the updates @Priva28! 👍 
✅ Approved. Can be merged as soon as the parent PR gets merged in.

**Comment by Priva28** [comment]
Thank you. Appreciate the review!

**Comment by tjorim** [comment]
One typo in the example configuration.yaml for dimmable lights, it mentions `type: color` instead of `type: dimmable`.

File: source/_components/light.httplight.markdown [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+Your device should be configured to return a brightness value from 0-100 when `http://IP_ADDRESS/brightness` is called. Your device should change the brightness of the light when it receives `http://IP_ADDRESS/set/BRIGHTNESS_HERE`.
+
+Example configuration for adding a dimmable light to Home Assistant:
+```yaml
+# Example configuration.yaml entry for HTTP Light that is dimmable.
+light:
+- platform: httplight
+  name: Dimmable Light
+  host: IP_ADDRESS
+  type: color
```
Comment (tjorim) [inline source/_components/light.httplight.markdown:80]: Replace type: color with type: dimmable.

**Review by tjorim — CHANGES_REQUESTED** [review]
One typo in the example configuration.yaml for dimmable lights which mentions color as type instead of dimmable.

**Comment by frenck** [comment]
@Priva28 Could you take a look at the comment left by @tjorim? Thanks already 👍

**Pull request closed without merging**